.github/workflows/release.yaml

```diff
@@ -0,0 +1,152 @@
+name: Deploy GCP on Tagging
+
+on:
+  release:
+    types:
+      - published  # Runs only when a release is published
+
+permissions:  
+  contents: write  # Grant write permissions to GITHUB_TOKEN
+
+jobs:
+  validate_gradle_wrapper:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        timeout-minutes: 1
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      # Gradle Wrapper validation can be flaky
+      # https://github.com/gradle/wrapper-validation-action/issues/40
+      - name: Gradle Wrapper Validation
+        timeout-minutes: 1
+        uses: gradle/wrapper-validation-action@f9c9c575b8b21b6485636a91ffecd10e558c62f6
+
+#  check_secrets:
+#    environment: deployment
+#    permissions:
+#      contents: read
+#    runs-on: ubuntu-latest
+#    outputs:
+#      has-secrets: ${{ steps.check_secrets.outputs.defined }}
+#    steps:
+#      - id: check_secrets
+#        env:
+#          COINBASE_APP_ID: ${{ secrets.COINBASE_APP_ID }}
+#        if: "${{
+#          env.COINBASE_APP_ID != ''
+#        }}"
+#        run: echo "defined=true" >> $GITHUB_OUTPUT
+
+  release:
+    environment: deployment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+            fetch-depth: 0 # to fetch all commits
+      - name: Set up Google Cloud SDK
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: ${{ secrets.GCP_SERVICE_ACCOUNT_KEY }}
+
+      - name: Configure gsutil
+        run: gcloud auth activate-service-account --key-file <(echo '${{ secrets.GCP_SERVICE_ACCOUNT_KEY }}')
+
+      - name: Download file from GCS
+        run: gsutil -q cp gs://${{ secrets.GCP_PROJECT_ID_PROD }}-apt-packages/encrypted_gpg.kms encrypted_gpg.kms
+
+      - name: Decrypt file using KMS
+        run: |
+          gcloud kms decrypt \
+            --key gpg \
+            --keyring gpg \
+            --location global \
+            --plaintext-file private.pgp \
```

Comment:
This is a little bit risky, no? If the workflow is exploitable then the key could potentially be leaked.

Reply:
@y4ssi @daira I had to close this in favor of the new PR https://github.com/Electric-Coin-Company/zashi-android/pull/1786. Linking this comment to the new PR and merging it. We need these changes for the related F-Droid work on reproducible builds. The new PR is going to be merged to the new common `gr-fdroid` feature branch